### PR TITLE
Soft Deletion and Undo API Along with Test Cases

### PR DESCRIPTION
**Pull request by YashC2410**

### CODE REVIEW

File: src/main/java/com/searchservice/app/rest/ManageTableResource.java [hunk 1]
```diff
@@ -122,17 +128,37 @@ public ManageTableResource(ManageTableServicePort manageTableServicePort) {
...
+		if(tableDeleteServicePort.checkTableExistensce(tableName)) {
+		    ResponseDTO apiResponseDTO = tableDeleteServicePort.initializeTableDelete(clientid, tableName);
+		    if (apiResponseDTO.getResponseStatusCode() == 200) {
+			 return ResponseEntity.status(HttpStatus.OK).body(apiResponseDTO);
+		   } else {
+			 log.debug("Exception occurred: {}", apiResponseDTO);
+			 throw new BadRequestOccurredException(400, BAD_REQUEST_MSG);
+		   }
+		}else {
+			throw new BadRequestOccurredException(400, "Table "+tableName+" Does Not Exist");
```
Comment (skdhitman) [inline src/main/java/com/searchservice/app/rest/ManageTableResource.java:141]: To be accurate, we can put Table 'xyz' for clientId '123' Does Not Exist

**Review by skdhitman — COMMENTED** [review]
Whole implementation looks good to me provided, it's tested with swagger properly

File: src/main/java/com/searchservice/app/rest/ManageTableResource.java (hunk 1 above)
Comment (YashC2410) [reply src/main/java/com/searchservice/app/rest/ManageTableResource.java:141]: yes surely should i make the changes

Comment (skdhitman) [reply src/main/java/com/searchservice/app/rest/ManageTableResource.java]: yes

**Pull request merged**